crates/credential/cargo-credential-gnome-secret/build.rs

```diff
@@ -0,0 +1,3 @@
+fn main() {
+    pkg_config::probe_library("libsecret-1").unwrap();
```

Comment:
This is one of my bigger worries about shipping these executables with rust-lang/rust. We typically don't make any assumptions about the execution environment, but this execution will require `libsecret.so` or something like that to run the process, right?

Or is there a way we could ask this to link statically when we build distribution binaries?

Reply:
Although one though I had reading this, these crates should all be trivial to `cargo install`, so we could perhaps manage it that way?

Reply:
Yea, this is a dynamic dependency, and the user will have to install the appropriate package on their system. I would feel a little uneasy statically linking this, primarily since I'm not very familiar with the GNOME ecosystem, but also because I'm not really sure it would work. There is infrastructure (the "Secret Service" and the gnome-keyring) that the user needs to have installed, and I suspect that the library that talks to them needs to be somewhat compatible.

I'm also not sure at this point how feasible it is to build this on rust-lang/rust, since they tend to use old Linux distributions which can be painful to use. I haven't yet tried. If it is too difficult to build, then I think we'll just have to tell the user to build it themselves (with `cargo install`), in which case they'll have to install libsecret anyways.

My feeling is that this will be an issue for documentation (it can include some sample instructions on what to install for various distributions).

In general, I see these sample implementations as a very "experimental" status. I wanted to see what issues may crop up, and have that guide how to proceed. If they don't work for some users, then they just won't work. Perhaps others can step up to either provide patches, or publish their own implementations on crates.io. If they stop working and go unmaintained, I think it would be reasonable just to delete them. (I would hope it doesn't come to that, I just want to emphasize that I think there is some flexibility here.) I have no idea how many people use GNOME secret storage, perhaps nobody will. 😄

Reply:
Ah ok makes sense, if we don't ship these with rust-lang/rust (which I initially thought we would), then this seems fine to ignore. I'm curious then, though, if we ship via Cargo and `cargo install`, do we still need the `libexec` logic? Or should we just look in `$PATH`?

Reply:
I would like to ship these wrappers, just not the libsecret one.  I checked, and the Linux used on rust-lang/rust is too old, so I updated the documentation to indicate that it needs to be installed manually.